python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -211,40 +187,88 @@ def _prepare_chat_history_for_request(
             chat_history: The chat history to prepare.
             role_key: The key name for the role/author.
             content_key: The key name for the content/message.
+            stream: Whether the request is for a streaming chat.
 
         Returns:
             A tuple containing the prepared chat history and the first SYSTEM message content.
         """
         system_message_content = None
-        remaining_messages = []
-
+        remaining_messages: list[dict[str, Any]] = []
         system_message_found = False
         for message in chat_history.messages:
             # Skip system messages after the first one is found
             if message.role == AuthorRole.SYSTEM:
                 if not system_message_found:
                     system_message_content = message.content
                     system_message_found = True
-                continue
+            elif message.role == AuthorRole.TOOL:
+                # if tool result message isn't the most recent message, add it to the remaining messages
+                if not remaining_messages or remaining_messages[-1][role_key] != AuthorRole.USER:
+                    remaining_messages.append({
+                        role_key: AuthorRole.USER,
+                        content_key: [],
```

Comment:
Will Anthropic really let us send an empty content_key here?

Reply:
This content gets populated immediately with the first tool call result, and any subsequent tool call results